daemon/list.go

```diff
@@ -133,7 +133,18 @@ func (daemon *Daemon) Containers(config *ContainersConfig) ([]*types.Container,
 			ID:    container.ID,
 			Names: names[container.ID],
 		}
-		newC.Image = container.Config.Image
+
+		img, err := daemon.Repositories().LookupImage(container.Config.Image)
+		if err != nil {
+			// If the image can no longer be found by its original reference,
+			// it makes sense to show the ID instead of a stale reference.
+			newC.Image = container.ImageID
```

Comment:
Hmm, ok. Now I'm not quite sure I've fully understood the problem. Eager to check out what you've got.